apps/projects/app/components/Panel/NewProject/RepoSelector.js

```diff
@@ -83,50 +89,54 @@ class Repo extends React.Component {
 
   unselect = () => this.setState({})
 
+  repoList = (visibleRepos, repoArray) => visibleRepos.length > 0 ? (
+    <RadioList
+      items={repoArray}
+      selected={this.state.repoSelected}
+      onChange={this.onRepoSelected}
+    />
+  ) : (
+    <div
+      style={{
+        display: 'flex',
+        alignItems: 'center',
+        marginTop: '19px',
+        flexDirection: 'column',
+      }}
+    >
+      <Text.Block>
+        There are no repositories matching{' '}
+        <Text weight="bold">{this.state.filter}</Text>
+      </Text.Block>
+      <ClearSearch onClick={this.handleClearSearch}>
+        Clear Search
+      </ClearSearch>
+    </div>
+  )
+
+  stillLoading = () => !(this.props.data && this.props.data.viewer && this.props.data.viewer.repositories.edges.length > 0)
+
   render() {
-    const { repos, filteredRepos, filtered, filter } = this.state
+    const { repos, filteredRepos, filtered, filter, reposAlreadyAdded } = this.state
 
     const visibleRepos = filtered ? filteredRepos : repos
-
     // TODO: extract to component
     const repoArray = visibleRepos.map(repo => ({
       title: repo.node.nameWithOwner,
       value: repo.node.id,
       description: '',
     }))
 
-    const repoList =
-      visibleRepos.length > 0 ? (
-        <RadioList
-          items={repoArray}
-          selected={this.state.repoSelected}
-          onChange={this.onRepoSelected}
-        />
-      ) : (
-        // <RadioGroup>
-        //   {visibleRepos.map((repo, i) => (
-        //     <RepoLine repo={repo} i={i} onRepoSelected={this.onRepoSelected} />
-        //   ))}
-        // </RadioGroup>
-        <div
-          style={{
-            display: 'flex',
-            alignItems: 'center',
-            marginTop: '19px',
-            flexDirection: 'column',
-          }}
-        >
-          <Text.Block>
-            There are no repositories matching{' '}
-            <Text weight="bold">{this.state.filter}</Text>
-          </Text.Block>
-          <ClearSearch onClick={this.handleClearSearch}>
-            Clear Search
-          </ClearSearch>
-        </div>
-      )
+    // if there are repos to add - list them. otherwise either all repos
+    // are already added or we're still waiting for the list from external
+    let repositoriesDisplay
 
-    const repositories = repos.length > 0 ? repoList : <Loading />
+    if (repos.length > 0)
+      repositoriesDisplay = this.repoList(visibleRepos, repoArray)
+    else if (this.stillLoading())
```

Comment:
It maybe should be better to use[ Apollo loading state flow](https://www.apollographql.com/docs/react/essentials/queries.html#loading) here

Reply:
Done.